### DISCUSSION TITLE
is `@hookform/resolvers@2.8` updated to match `react-hook-form@7.18.2` ?

### DISCUSSION THREAD

**Discussion by scr2em** [react-hook-form/react-hook-form#6491] [Q&A]
Hi, I got types error when I migrated to `react-hook-form@7.15.1` in my YupResolver ? I updated my `@hookform/resolvers` to the last version and I still encounter the errors
```
"dependencies": {
    "@hookform/resolvers": "^2.8.0",
      "react-hook-form": "^7.18.2",
        "yup": "^0.32.9"
  },
  "devDependencies": {
       "@types/yup": "^0.29.12",
        "typescript": "^4.3.5"
  },
  ```

**Comment by bluebill1049** [comment]
what error are you getting?

**Reply by scr2em** [reply]
it was working ok, till I upgraded to 7.18.2
![image](https://user-images.githubusercontent.com/4671486/132493999-9adfa840-fb6b-4f35-9190-2f991ce406bb.png)

**Reply by bluebill1049** [reply]
can you share your `type` at useForm?

**Reply by bluebill1049** [reply]
i suspect its due to the union, can you reproduce this in a codesandbox?

**Reply by bluebill1049** [reply]
cc @kotarella1110
